app/src/main/java/org/wikipedia/editactionfeed/MyContributionsFragment.kt

```diff
@@ -0,0 +1,129 @@
+        inflater!!.inflate(R.menu.menu_edit_tasks, menu)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
+        return when (item!!.itemId) {
+            R.id.menu_help -> {
+                FeedbackUtil.showAndroidAppEditingFAQ(requireContext())
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    private fun updateUI() {
+        requireActivity().invalidateOptionsMenu()
+        fetchUserContributions()
+    }
+
+    private fun fetchUserContributions() {
+        contributionsText.visibility = View.GONE
+        progressBar.visibility = View.VISIBLE
+
+        disposables.add(ServiceFactory.get(WikiSite(Service.WIKIDATA_URL)).editorTaskCounts
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .doFinally {
+                    progressBar.visibility = View.GONE
+                    contributionsText.visibility = View.VISIBLE
+                    swipeRefreshLayout.isRefreshing = false
+                }
+                .subscribe({ response ->
+                    editorTaskCounts = response.query()!!.editorTaskCounts()!!
+                    NotificationEditorTasksHandler.dispatchEditorTaskResults(requireContext(), editorTaskCounts)
+                    val totalEdits = editorTaskCounts.descriptionEditsPerLanguage!!.values.sum()
+                    languageList = editorTaskCounts.descriptionEditsPerLanguage!!.keys.toList()
+                    contributionsText.text = resources.getQuantityString(R.plurals.edit_action_contribution_count, totalEdits, totalEdits)
+                    adapter.notifyDataSetChanged()
+                }, { throwable ->
+                    L.e(throwable)
+                    FeedbackUtil.showError(requireActivity(), throwable)
+                }))
+    }
+
+    private inner class MyContributionsItemAdapter : RecyclerView.Adapter<ItemViewHolder>() {
+
+        override fun getItemCount(): Int {
+            return languageList.size
+        }
+
+        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemViewHolder {
+            return ItemViewHolder(LayoutInflater.from(parent.context)
+                    .inflate(R.layout.item_my_contributions, parent, false))
+        }
+
+        override fun onBindViewHolder(holder: ItemViewHolder, pos: Int) {
+            holder.bindItem(languageList[pos])
+        }
+    }
+
+    private inner class ItemViewHolder internal constructor(itemView: View) : DefaultViewHolder<View>(itemView) {
+        internal fun bindItem(langCode: String) {
+            ViewUtil.formatLangButton(itemView.languageCode, langCode, LANG_BUTTON_TEXT_SIZE_SMALLER, LANG_BUTTON_TEXT_SIZE_LARGER)
+            itemView.languageCode.text = langCode
+            itemView.languageTitle.text = WikipediaApp.getInstance().language().getAppLanguageLocalizedName(langCode)
```

Comment:
It needs to use Canonical name : https://app.zeplin.io/project/57a120b91998d8977642a238/screen/5c77dc34d693e3bc7268a88b